router/batchrouter/asyncdestinationmanager/bing-ads/util.go

```diff
@@ -0,0 +1,444 @@
+package bingads
+
+import (
+	"archive/zip"
+	"bufio"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+// Upload related utils
+
+// returns the clientID struct
+func newClientID(jobID int64, hashedEmail string) ClientID {
+	return ClientID{
+		JobID:       jobID,
+		HashedEmail: hashedEmail,
+	}
+}
+
+// returns the string representation of the clientID struct which is of format
+// jobId<<>>hashedEmail
+func (c *ClientID) ToString() string {
+	return fmt.Sprintf("%d%s%s", c.JobID, clientIDSeparator, c.HashedEmail)
+}
+
+/*
+returns the csv file and zip file path, along with the csv writer that
+contains the template of the uploadable file.
+*/
+func createActionFile(audienceId, actionType string) (*ActionFileInfo, error) {
```

Comment:
Do we have tests for these functions? Are they covered in any of the test flow?

Reply:
yes they are